src/components/lab/EmittedLine.js

```diff
@@ -29,24 +30,35 @@ export default class EmittedLine extends Component {
     }).isRequired,
     numberOfLines: PropTypes.number.isRequired,
     maxPointsForLines: PropTypes.number.isRequired,
+    oscillation: PropTypes.bool.isRequired,
+    timerCount: PropTypes.number.isRequired,
   };
 
   componentDidMount() {
     this.beginLineInterval();
   }
 
   componentDidUpdate(prevProps, prevState) {
-    const { isPaused, numberOfLines } = this.props;
-    if (isPaused !== prevProps.isPaused && isPaused) {
+    const { isPaused, oscillation, timerCount, numberOfLines } = this.props;
+    if (isPaused !== prevProps.isPaused && isPaused && !oscillation) {
+      clearInterval(this.emittedLineInterval);
+    } else if (isPaused !== prevProps.isPaused && !isPaused && oscillation) {
```

Comment:
Maybe having `isPaused !== prevProps.isPaused` in a common if-clause would be clearer.